File: packages/expo-sensors/android/src/main/java/expo/modules/sensors/services/LinearAccelerationSensorService.java
// Copyright 2015-present 650 Industries. All rights reserved.

package expo.modules.sensors.services;

import android.content.Context;
import android.hardware.Sensor;

import java.util.Collections;
import java.util.List;

import org.unimodules.core.interfaces.InternalModule;

public class LinearAccelerationSensorService extends SubscribableSensorService implements InternalModule, org.unimodules.interfaces.sensors.services.LinearAccelerationSensorService {
  public LinearAccelerationSensorService(Context reactContext) {
    super(reactContext);
  }

  @Override
  int getSensorType() {
    return Sensor.TYPE_LINEAR_ACCELERATION;
  }

  @Override
  public List<Class> getExportedInterfaces() {
    return Collections.<Class>singletonList(org.unimodules.interfaces.sensors.services.LinearAccelerationSensorService.class);
  }
}

File: packages/expo-sensors/android/src/main/java/expo/modules/sensors/services/LinearAccelerationSensorService.kt
// Copyright 2015-present 650 Industries. All rights reserved.
package expo.modules.sensors.services

import android.content.Context
import android.hardware.Sensor
import expo.modules.interfaces.sensors.services.LinearAccelerationSensorServiceInterface
import expo.modules.core.interfaces.InternalModule

class LinearAccelerationSensorService(reactContext: Context?) : SubscribableSensorService(reactContext), InternalModule, LinearAccelerationSensorServiceInterface {
  override val sensorType: Int = Sensor.TYPE_LINEAR_ACCELERATION

  override fun getExportedInterfaces(): List<Class<*>> {
    return listOf<Class<*>>(LinearAccelerationSensorServiceInterface::class.java)
  }
}
